n26/api.py

```diff
@@ -420,12 +420,17 @@ def _refresh_token(refresh_token: str):
         return response.json()
 
     @staticmethod
-    def _request_mfa_approval(mfa_token: str):
+    def _request_mfa_approval(self, mfa_token: str):
         LOGGER.debug("Requesting MFA approval using mfa_token {}".format(mfa_token))
         mfa_data = {
-            "challengeType": "oob",
             "mfaToken": mfa_token
         }
+
+        if self.config.mfa_type == "otp":
+            mfa_data['challengeType'] = "otp"
+        else:
+            mfa_data['challengeType'] = "oob"
```

Comment:
Please add constants for `otp` and `oob` in the `const.py` and import them here **when you compare the config value**. That way we can be sure there are no typos when comparing the config value. You can still use bare strings when setting the `mfa_data` dict though as they are independent from the config value.

Reply:
If you have done this please also add a validation to the config that check if the given value is one of the valid ones.

Reply:
You may have a better idea of how this should look. I will hand this over to you so I don't butcher anything.

Reply:
Done.